### PR TITLE
COMMERCE-1496 CommercePiceList replace commerceCurrencyId with commerceCurrencyCode

### PR DESCRIPTION
**Pull request by luca-pellizzon**
@marco-leo

### CODE REVIEW

**Comment by ethanbustad** [comment]
ci:test

**Comment by marco-leo** [comment]
@luca-pellizzon let's rebase this, I would like to get in 2.0.7

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
@ethanbustad I think we need to update our local SF since all of these errors are not showing when running locally

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by ethanbustad** [comment]
@luca-pellizzon yeah, I'll look into why this is happening. I did just update SF, but we've always had some checks that just don't seem to run locally, so I'll try to figure that out. For this sorting issue in particular, it wants the setters to be in the same order as the fields are declared in `service.xml`, FYI.

**Comment by ethanbustad** [comment]
Hey @luca-pellizzon , I don't see any reason why this check isn't running locally... But we're going to be getting rid of the subrepo soon, and this discrepancy shouldn't be possible any more once we make the switch. So I just fixed it in upstream this time, and hopefully this is the last time!

**Comment by ethanbustad** [comment]
Rebased and sent to https://github.com/liferay/com-liferay-commerce/pull/1467 since my fix caused a conflict here.

**Comment by ethanbustad** [comment]
Sorry, @luca-pellizzon , I left some review comments at https://github.com/liferay/com-liferay-commerce/pull/1467#pullrequestreview-296477550. It's probably easier to just make the changes here and force-push the branch, so reopening this PR. The only change I made other than rebasing was to remove the last `source formatting` commit.

**Review by ethanbustad — CHANGES_REQUESTED** [review]
Bah, requires a comment. See above.

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by marco-leo** [comment]
@luca-pellizzon let's fix the SF please

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by ethanbustad** [comment]
@luca-pellizzon looks like the error is in the sorting of setters. In `*LocalServiceImpl.java`, setters should be sorted in the same order as the fields are declared in `service.xml`.

**Comment by luca-pellizzon** [comment]
ci:test

File: commerce-price-list-service/src/main/java/com/liferay/commerce/price/list/internal/upgrade/v3_0_0/CommercePriceListUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ */
+public class CommercePriceListUpgradeProcess
+	extends BaseCommercePriceListUpgradeProcess {
+
+	@Override
+	public void doUpgrade() throws Exception {
+
+		// Add commerceCurrencyCode column
+
+		String newColumnName = "commerceCurrencyCode";
```
Comment (marco-leo) [inline commerce-price-list-service/src/main/java/com/liferay/commerce/price/list/internal/upgrade/v3_0_0/CommercePriceListUpgradeProcess.java]: @luca-pellizzon we can avoid this variable

Comment (luca-pellizzon) [reply commerce-price-list-service/src/main/java/com/liferay/commerce/price/list/internal/upgrade/v3_0_0/CommercePriceListUpgradeProcess.java]: updated

**Comment by marco-leo** [comment]
sent to https://github.com/liferay/com-liferay-commerce/pull/1511

**Pull request closed without merging**